Content.Server/GameObjects/Components/ActionBlocking/CuffedComponent.cs

```diff
@@ -0,0 +1,352 @@
+        /// </summary>
+        /// <param name="user">The cuffed entity</param>
+        /// <param name="cuffsToRemove">Optional param for the handcuff entity to remove from the cuffed entity. If null, uses the most recently added handcuff entity.</param>
+        public async void TryUncuff(IEntity user, IEntity cuffsToRemove = null)
+        {
+            var isOwner = user == Owner;
+
+            if (cuffsToRemove == null)
+            {
+                cuffsToRemove = LastAddedCuffs;
+            }
+            else
+            {
+                if (!_container.ContainedEntities.Contains(cuffsToRemove))
+                {
+                    Logger.Warning($"A user is trying to remove handcuffs that aren't in the owner's container. This should never happen!");
+                }
+            }
+
+            if (!cuffsToRemove.TryGetComponent<HandcuffComponent>(out var cuff))
+            {
+                Logger.Warning($"A user is trying to remove handcuffs without a ${nameof(HandcuffComponent)}. This should never happen!");
+                return;
+            }
+
+            if (!isOwner && !ActionBlockerSystem.CanInteract(user))
+            {
+                _notifyManager.PopupMessage(user, user, "You can't do that!");
+                return;
+            }
+
+            if (!isOwner &&
+                !EntitySystem.Get<SharedInteractionSystem>().InRangeUnobstructed(
+                    user.Transform.MapPosition,
+                    Owner.Transform.MapPosition,
+                    _interactRange,
+                    ignoredEnt: Owner))
+            {
+                _notifyManager.PopupMessage(user, user, "You are too far away to remove the cuffs.");
+                return;
+            }
+
+            if (!EntitySystem.Get<SharedInteractionSystem>().InRangeUnobstructed(
+                    cuffsToRemove.Transform.MapPosition,
+                    Owner.Transform.MapPosition,
+                    _interactRange,
+                    ignoredEnt: Owner))
+            {
+                Logger.Warning($"Handcuffs being removed from player are obstructed or too far away! This should not happen!");
+                return;
+            }
+
+            _notifyManager.PopupMessage(user, user, "You start removing the cuffs.");
+            _audioSystem.PlayFromEntity(isOwner ? cuff.StartBreakoutSound : cuff.StartUncuffSound, Owner);
+
+            var uncuffTime = isOwner ? cuff.BreakoutTime : cuff.UncuffTime;
+            var doAfterEventArgs = new DoAfterEventArgs(user, uncuffTime)
+            {
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                NeedHand = true
+            };
+
+            var result = await _doAfterSystem.DoAfter(doAfterEventArgs);
+
+            if (result != DoAfterStatus.Cancelled)
+            {
+                _audioSystem.PlayFromEntity(cuff.EndUncuffSound, Owner);
```

Comment:
You can just use the static function EntitySystem.Get<AudioSystem> for these and not need the member variable, same with DoAfterSystem I guess.